File: Code/web_admin/src/mock/share.js
import Mock from 'mockjs';

const shareApi = {
  getShareList: "/api/admin/share/list/",
  getShareInfo: "/api/admin/share/",
  deleteShare: "/api/admin/share/",
};

const shareList = [{
  "id": "test",
  "userId": 1,
  "repoId": "5f0c3139b9555b210e4d056c",
  "name": "test",
  "password": "123456",
  "status": 1,
  "createTime": 1594634553484,
  "validTime": 1594634553484
}];

const share = {
  "share": {
    "id": "test",
    "userId": 1,
    "repoId": "5f0c3139b9555b210e4d056e",
    "name": "test",
    "password": "123456",
    "status": 1,
    "createTime": 1594634553728,
    "validTime": 1594634553728
  },
  "shareRepository": {
    "id": "5f0c3139b9555b210e4d056e",
    "shareId": "test",
    "userId": 1,
    "userFileIdList": [],
    "fileIdList": [],
    "saveUserIdList": {},
    "downloadUserIdList": {},
    "folder": {
      "name": null,
      "path": null,
      "depth": 0,
      "createTime": 0,
      "changeTime": 0,
      "status": 0,
      "folders": null,
      "files": null
    }
  }
};


Mock.mock(shareApi.getShareList + 1, "get", {code: 1, message: "查询成功", data: {shareList: shareList}});
Mock.mock(shareApi.getShareInfo + "test" + "/info", "get", {
  code: 1,
  message: "查询成功",
  data: share
});
Mock.mock(shareApi.deleteShare + "test", "delete", {code: 1, message: "删除成功", data: null});

File: Code/cloud/src/main/java/com/ztu/cloud/cloud/common/bean/mysql/Share.java
package com.ztu.cloud.cloud.common.bean.mysql;

import lombok.Data;

/**
 * @author Jager
 * @description 分享信息表
 * @date 2020/06/21-12:46
 **/
@Data
public class Share {
    private String id;
    private int userId;
    private String repoId;
    private String name;
    private String password;
    private int status;
    private long createTime;
    private long validTime;

    public Share() {
    }

    public Share(String id, int userId, String repoId, String name, String password, int status, long createTime, long validTime) {
        this.id = id;
        this.userId = userId;
        this.repoId = repoId;
        this.name = name;
        this.password = password;
        this.status = status;
        this.createTime = createTime;
        this.validTime = validTime;
    }

    public Share(String id, int userId, String repoId, String name, String password, long validTime) {
        this.id = id;
        this.userId = userId;
        this.repoId = repoId;
        this.name = name;
        this.password = password;
        this.status = 1;
        this.createTime = System.currentTimeMillis();
        this.validTime = validTime;
    }
}
